pkg/network/discovery.go

```diff
@@ -144,26 +150,33 @@ func (d *DefaultDiscovery) RequestRemote(requested int) {
 	}
 }
 
-// RegisterBadAddr registers the given address as a bad address.
-func (d *DefaultDiscovery) RegisterBadAddr(addr string) {
-	var isSeed bool
+// RegisterSelf registers the given Peer as a bad one, because it's our own node.
+func (d *DefaultDiscovery) RegisterSelf(p AddressablePeer) {
+	var connaddr = p.ConnectionAddr()
 	d.lock.Lock()
-	for _, seed := range d.seeds {
-		if addr == seed {
-			isSeed = true
-			break
+	delete(d.connectedAddrs, connaddr)
+	d.registerBad(connaddr, true)
+	d.registerBad(p.PeerAddr().String(), true)
+	d.lock.Unlock()
+}
+
+func (d *DefaultDiscovery) registerBad(addr string, force bool) {
+	_, isSeed := d.seeds[addr]
```

Comment:
Don't  lock seeds before access?

Reply:
It's an internal function, all users of it take a lock before going here.